typescript/src/board.ts

```diff
@@ -2,9 +2,15 @@
 import * as _ from 'lodash';
 import * as orbs from './orbs';
 import * as triples from './triples';
+import * as types from '../types';
 
 export class Board {
-    constructor (width = 8, height = 8, types = _.range(7), needsAttic = true) {
+    width: number;
+    height: number;
+    types: any[]
+    orbs: any[][];
+    attic: Board;
+    constructor (width: number = 8, height: number = 8, types: any[] = _.range(7), needsAttic: boolean = true) {
```

Comment:
I thought quite a bit about how to handle the type of `this.types`. I wanted to create a generic type in the function that matched whatever `typeof this.types` is. Basically, you can't do that is what I learned (I asked this [question](http://stackoverflow.com/questions/38232987/use-parameter-as-a-generic-type-and-also-give-it-a-default-assignment/38233862?noredirect=1#comment63892180_38233862)). Ultimately, I went with just calling it `any[]` since something like `[1, 2, 'a', 'b', null]` could technically work as a set of types.